components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AuthenticationResponseProcessor.java

```diff
@@ -0,0 +1,244 @@
+                    for (PerformableOperation operation : operationsToPerform) {
+                        performOperation(operation, authRequestUser, userStore);
+                    }
+                }
+                context.setSubject(createAuthenticateduser(authRequestUser, context, userStore));
+            } catch (UserStoreException e) {
+                throw new ActionExecutionResponseProcessorException("Error occurred when trying to build authenticated " +
+                        "user from the external authenticator service response." ,e);
+            }
+        }
+
+        // TODO: Log operation process results.
+        return new ActionExecutionStatus(ActionExecutionStatus.Status.SUCCESS, eventContext);
+    }
+
+    @Override
+    public ActionExecutionStatus processErrorResponse(Map<String, Object> eventContext,
+                                                      Event actionEvent,
+                                                      ActionInvocationErrorResponse errorResponse) throws
+            ActionExecutionResponseProcessorException {
+
+        return new ActionExecutionStatus(ActionExecutionStatus.Status.ERROR, eventContext);
+    }
+
+    /**
+     * Validate the given performableOperation and apply it to the user attribute.
+     *
+     * @param performableOperation  PerformableOperation from the external authentication service response.
+     */
+    public void performOperation(PerformableOperation performableOperation, AuthenticationRequestUser authRequestUser,
+                                 UserStore userStoreDomain) throws ActionExecutionResponseProcessorException {
+
+        if (!isAllowedOperation(performableOperation.getOp())) {
+            return;
+        }
+
+        switch (performableOperation.getPath()) {
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.USER_ID_PATH:
+                authRequestUser.setId(castToString(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.EXTERNAL_ID_PATH:
+                authRequestUser.setExternalId(castToString(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.IDP_PATH:
+                authRequestUser.setIdp(castToString(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.SUB_PATH:
+                authRequestUser.setSub(castToString(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.USER_CLAIM_PATH:
+                authRequestUser.setUserClaims(castToClaim(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.USER_STORE_NAME_PATH:
+                userStoreDomain.setName(castToString(performableOperation.getValue()));
```

Comment:
Does each of these adhere's to the input validations the server expects ?

Reply:
When creating AuthenticatedUser from these input, validation happens with the AuthenticatedUserBuilder.java.